lib/internal/Magento/Framework/Stdlib/DateTime/Timezone.php

```diff
@@ -274,7 +274,7 @@ public function isScopeDateInInterval($scope, $dateFrom = null, $dateTo = null)
         $scopeTimeStamp = $this->scopeTimeStamp($scope);
         $fromTimeStamp = strtotime($dateFrom);
         $toTimeStamp = strtotime($dateTo);
-        if ($dateTo) {
+        if ($dateTo && date('H:i:s', strtotime($dateTo)) === '00:00:00') {
```

Comment:
We will have the same +1 day result if use '00:00:00' in the API query. I'm not sure that it's correct solution